org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -1336,6 +1338,19 @@ public Preferences setSettingsUrl(String settingsUrl) {
 
 	public Preferences setResourceFilters(List<String> resourceFilters) {
 		this.resourceFilters = resourceFilters == null ? new ArrayList<>() : resourceFilters;
+		if (resourceFilters != null) {
+			this.resourceFilters = resourceFilters.stream().filter((resource) -> {
+				try {
+					Pattern.compile(resource);
+					return true;
+				} catch (Exception e) {
+					JavaLanguageServerPlugin.logInfo("Invalid preference: " + Preferences.JAVA_RESOURCE_FILTERS + "=" + resource);
+					return false;
+				}
+			}).collect(Collectors.toList());
+		} else {
+			this.resourceFilters = resourceFilters;
```

Comment:
Remove line 1340. And assign `this.resourceFilters` to empty list if the input argument is null.

Reply:
Fixed.